### ISSUE TITLE
[@angular-eslint/template/no-nested-tags] false positive

### ISSUE THREAD

**Issue by splincode** [angular-eslint/angular-eslint#3075]
I have nested `ng-template` https://github.com/taiga-family/taiga-ui/blob/main/projects/demo/src/pages/directives/dropdown-hover/examples/3/index.html#L41

But 

```
/projects/demo/src/pages/directives/dropdown-hover/examples/3/index.html
  41:21  error  <p> elements must not be nested! This breaks angular incremental hydration as all browsers will convert "<p>1<p>2</p>3</p>" into "<p>1</p><p>2</p>3", creating a DOM mismatch between SSR and Angular  @angular-eslint/template/no-nested-tags
```

I think it's false positive:

```html
<p>
  Text
   <ng-template #content>
     <p>Text</p>
   </ng-template>
</p>
```

**Comment by morgan-coded** [comment]
I can take this if it’s still available. I’d keep the change narrow: treat a real `<ng-template>` as a boundary for `no-nested-tags`, while keeping structural directive cases like `*ngFor` reportable.

**Comment by morgan-coded** [comment]
Re-checked this at the current 22.1.0 head — the false positive still reproduces, and the narrow fix I offered in June still applies cleanly: treat a real `<ng-template>` as a boundary in `no-nested-tags`, while same-tag nesting through structural directives like `*ngFor` keeps reporting. Happy to open the PR if that direction works, or to drop it if you'd rather go another way.
